libbeat/autodiscover/autodiscover_test.go

```diff
@@ -289,6 +298,67 @@ func TestAutodiscoverHash(t *testing.T) {
 	assert.False(t, runners[1].stopped)
 }
 
+func TestAutodiscoverWithConfigCheckFailures(t *testing.T) {
+	// Register mock autodiscover provider
+	busChan := make(chan bus.Bus, 1)
+	Registry = NewRegistry()
+	Registry.AddProvider("mock", func(b bus.Bus, c *common.Config) (Provider, error) {
+		// intercept bus to mock events
+		busChan <- b
+
+		return &mockProvider{}, nil
+	})
+
+	adapter := mockAdapter{
+		configs: []*common.Config{},
+	}
+
+	// and settings:
+	providerConfig, _ := common.NewConfigFrom(map[string]string{
+		"type": "mock",
+	})
+	config := Config{
+		Providers: []*common.Config{providerConfig},
+	}
+
+	// Create autodiscover manager
+	autodiscover, err := NewAutodiscover("test", nil, &adapter, &config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Start it
+	autodiscover.Start()
+	defer autodiscover.Stop()
+	eventBus := <-busChan
+
+	// Test start event
+	eventBus.Publish(bus.Event{
+		"start": true,
+		"meta": common.MapStr{
+			"foo": "bar",
+		},
+	})
+
+	// Timed sleep to make sure that bus event is processed
+	time.Sleep(time.Second)
```

Comment:
maybe you can use the `wait` helper here?

Reply:
@exekias done